File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
[10,1,4,3,6,1, 7,1]

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2:
            return 0
        
        max_price = 0
        

        l = 0
        r = 1


        while l < r and r < len(prices):
            if prices[l] > prices[r]:
                l = r
                r += 1
                continue
            
            max_price = max(max_price, prices[r] - prices[l])
            r += 1
        

        return max_price

            
            







            






            

File: Data Structures & Algorithms/house-robber/submission-2.cs
    public class Solution
    {
        public int Rob(int[] nums)
        {
            if(nums.Length == 1) return nums[0];

            int[] opt = new int[nums.Length];
            
            opt[0] = nums[0];
            opt[1] = Math.Max(nums[0], nums[1]);


            for (int i = 2; i < nums.Length; i++)
            {
                //#1 take current and add it to the i - 2
                //#2 dont take current, so copy from oi - 1

                if (nums[i] + opt[i - 2] > opt[i - 1])
                {
                    opt[i] = nums[i] + opt[i - 2];
                }
                else
                {
                    opt[i] = opt[i - 1];
                }
            }

            return opt.Last();
        }
    }

